doc/specs/#888 - Com Api.md

```diff
@@ -0,0 +1,812 @@
+        filter.Field = PackageMatchField::Id;
+        filter.Type = MatchType::Exact;
+        filter.Value = packageId;
+        findPackagesOptions.Filters().Append(filter);
+        FindPackagesResult findPackagesResult{ co_await catalog.FindPackagesAsync(findPackagesOptions) };
+
+        winrt::IVectorView<ResultMatch> matches = findPackagesResult.Matches();
+        co_return matches.GetAt(0).Package();
+    }
+
+    IAsyncOperation<Package> MainPage::FindPackage()
+    {
+        // Capture the ui thread context.
+        winrt::apartment_context uiThread;
+        co_await winrt::resume_background();
+
+        AppInstaller appInstaller = CreateAppInstaller();
+        AppCatalog catalog{ co_await appInstaller.GetAppCatalogAsync(PredefinedAppCatalog::OpenWindowsCatalog) };
+        co_await catalog.OpenAsync();
+        Package package = { co_await FindPackageInCatalog(catalog, m_installAppId) };
+
+        co_return package;
+    }
+
+```
+
+## 3.3. Install
+
+```c++ (C++ish pseudocode)
+
+    IAsyncOperationWithProgress<InstallResult, InstallProgress> InstallPackage(Package package)
+    {
+        AppInstaller appInstaller = CreateAppInstaller();
+        InstallOptions installOptions = CreateInstallOptions();
+
+        installOptions.InstallScope(InstallScope::User);
+        installOptions.Package(package);
+        installOptions.InstallMode(InstallMode::Silent);
+
+        return appInstaller.InstallPackageAsync(installOptions);
+    }
+
+    IAsyncAction UpdateUIProgress(InstallProgress progress, winrt::apartment_context uiThread, winrt::Windows::UI::Xaml::Controls::ProgressBar progressBar, winrt::Windows::UI::Xaml::Controls::TextBlock statusText)
+    {
+        co_await uiThread;
+        progressBar.Value(progress.DownloadPercentage);
+
+        std::wstring downloadText{ L"Downloading." };
+        switch (progress.State)
+        {
+        case InstallProgressState::Queued:
+            statusText.Text(L"Queued");
+            break;
+        case InstallProgressState::Downloading:
+            downloadText += progress.BytesDownloaded + L" bytes of " + progress.BytesRequired;
+            statusText.Text(downloadText.c_str());
+            break;
+        case InstallProgressState::Installing:
+            statusText.Text(L"Installing");
+            break;
+        case InstallProgressState::PostInstall:
+            statusText.Text(L"Finishing install");
+            break;
+        default:
+            statusText.Text(L"");
+        }
+        co_return;
+    }
+
+    IAsyncAction UpdateUIForInstall(
+        IAsyncOperationWithProgress<InstallResult, InstallProgress> installPackageOperation, 
+        winrt::apartment_context uiThread,
+        winrt::Windows::UI::Xaml::Controls::Button button,
+        winrt::Windows::UI::Xaml::Controls::ProgressBar progressBar, 
+        winrt::Windows::UI::Xaml::Controls::TextBlock statusText)
+    {
+        installPackageOperation.Progress([&](

```

Comment:
Capturing by reference here seems like it could be a problem; the `co_await` on line 165 effectively returns from this block, invalidating the captured-by-reference things that were passed in.  Consider capturing by `[=]` instead.